pkg/tsdb/influxdb/influxql/util/util.go

```diff
@@ -0,0 +1,154 @@
+package util
+
+import (
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+
+	"github.com/grafana/grafana/pkg/tsdb/influxdb/models"
+)
+
+var (
+	legendFormat = regexp.MustCompile(`\[\[([\@\/\w-]+)(\.[\@\/\w-]+)*\]\]*|\$([\@\w-]+?)*`)
+)
+
+const (
+	GraphVisType data.VisType = "graph"
+	TableVisType data.VisType = "table"
+	LogsVisType  data.VisType = "logs"
+)
+
+func FormatFrameName(rowName, column string, tags map[string]string, query models.Query, frameName []byte) []byte {
+	if query.Alias == "" {
+		return BuildFrameNameFromQuery(rowName, column, tags, frameName, query.ResultFormat)
+	}
+	nameSegment := strings.Split(rowName, ".")
+
+	result := legendFormat.ReplaceAllFunc([]byte(query.Alias), func(in []byte) []byte {
+		aliasFormat := string(in)
+		aliasFormat = strings.Replace(aliasFormat, "[[", "", 1)
+		aliasFormat = strings.Replace(aliasFormat, "]]", "", 1)
+		aliasFormat = strings.Replace(aliasFormat, "$", "", 1)
+
+		if aliasFormat == "m" || aliasFormat == "measurement" {
+			return []byte(rowName)
+		}
+		if aliasFormat == "col" {
+			return []byte(column)
+		}
+
+		pos, err := strconv.Atoi(aliasFormat)
+		if err == nil && len(nameSegment) > pos {
+			return []byte(nameSegment[pos])
+		}
+
+		if !strings.HasPrefix(aliasFormat, "tag_") {
+			return in
+		}
+
+		tagKey := strings.Replace(aliasFormat, "tag_", "", 1)
+		tagValue, exist := tags[tagKey]
+		if exist {
+			return []byte(tagValue)
+		}
+
+		return in
+	})
+
+	return result
+}
+
+func BuildFrameNameFromQuery(rowName, column string, tags map[string]string, frameName []byte, resultFormat string) []byte {
+	if resultFormat != "table" {
+		frameName = append(frameName, rowName...)
+		frameName = append(frameName, '.')
+	}
+	frameName = append(frameName, column...)
+
+	if len(tags) > 0 {
+		frameName = append(frameName, ' ', '{', ' ')
+		first := true
+		for k, v := range tags {
+			if !first {
+				frameName = append(frameName, ',')
+				frameName = append(frameName, ' ')
+			} else {
+				first = false
+			}
+			frameName = append(frameName, k...)
+			frameName = append(frameName, ':', ' ')
+			frameName = append(frameName, v...)
+		}
+
+		frameName = append(frameName, ' ', '}')
+	}
+
+	return frameName
```

Comment:
```suggestion
	if len(tags) == 0 {
		return frameName
	}
	frameName = append(frameName, ' ', '{', ' ')
	first := true
	for k, v := range tags {
		if !first {
			frameName = append(frameName, ',')
			frameName = append(frameName, ' ')
		} else {
			first = false
		}
		frameName = append(frameName, k...)
		frameName = append(frameName, ':', ' ')
		frameName = append(frameName, v...)
	}
	return append(frameName, ' ', '}')
```

Reply:
I have this in https://github.com/grafana/grafana/pull/78826